x-pack/plugins/alerting/server/rule_type_registry.ts

```diff
@@ -420,6 +420,10 @@ export class RuleTypeRegistry {
   public getAllTypes(): string[] {
     return [...this.ruleTypes.keys()];
   }
+
+  public async createAlertsClient(...args: Parameters<AlertsService['createAlertsClient']>) {
```

Comment:
Drive-by comment. I think it would make more sense to pass in the AlertsService to the RulesClientFactory -> RulesClient and use it directly there to create the alerts client rather than adding this pass through function in the rule type registry

Reply:
Yeah that makes sense. This PR has been ballooning out of control with so many different services and plugins getting added to new things, and core types getting modified, and the number of file changes exploding, that I've been doing everything I can to avoid not adding another thing that's going to trigger a typecheck failure across an unpredictable number of test files, but I guess we've already crossed that threshold. My instinct as a developer to keep PRs small does not work with the reality of the Kibana codebase.

I'll get to it.